lib/mcb/grant_access_wizard.rb

```diff
@@ -57,14 +62,42 @@ def persist_user_if_new
     end
 
     def confirm_and_add_user_to_organisation
+      unless @provider
+        puts "\n"
+        puts "Error: specify a provider code"
+        return
+      end
+
+      @organisation = @provider.organisations.first # a provider should only ever be associated with one organisation
       if @user.in?(@organisation.users)
         puts "#{@user} already belongs to #{@organisation.name}"
-      else
-        puts "\n"
-        puts "You're about to give #{@user} access to organisation #{@organisation.name}. They will manage:"
-        puts MCB::Render::ActiveRecord.providers_table @organisation.providers, name: "Additional Providers"
-        @organisation.users << @user if @cli.agree("Agree?  ")
+        return
+      end
+
+      add_user_to_org
+    end
+
+    def add_user_to_org
+      puts "\n"
+      puts "You're about to give #{@user} access to organisation #{@organisation.name}. They will manage:"
+      puts MCB::Render::ActiveRecord.providers_table @organisation.providers, name: "Additional Providers"
+      @organisation.users << @user if @cli.agree("Agree?  ")
+    end
+
+    def confirm_and_add_user_to_all_organisations
+      unless @user.admin?
+        puts "Refusing to give non-admin user #{@user} access to all orgs"
```

Comment:
`@user` could be renamed to something more real?

Suggestions: `@user_to_update`, `@target_user` or `@user_to_grant`